README.md

```diff
@@ -159,15 +159,25 @@ any errors encountered during the request.
 
 ## Body
 
-If the server's response contains a body it should follow the requirements for [GraphQL response](https://graphql.github.io/graphql-spec/June2018/#sec-Response).
+If the server's response contains a body it MUST follow the requirements for [GraphQL response](https://graphql.github.io/graphql-spec/June2018/#sec-Response).
 
-Note: For any non-2XX response, the client should not rely on the body to be in GraphQL format since the source of the response
-may not be the GraphQL server but instead some intermediary such as API gateways, firewalls, etc.
+Note: For any non-2XX response, the client should not rely on the body to be in GraphQL format.
+The source of the response may not be the GraphQL server, but instead some intermediary such as API gateways, firewalls, etc.
 
 ## Status Codes
 
-If the `data` entry in the response has any value other than `null` (when the operation has successfully executed
-without error) then the response should use the 200 (OK) status code.
-If the operation failed before or during execution, due to a syntax error, missing information, validation error
-or any other reason, then response should use the 4XX or 5XX status codes.
-It is recommended to use the same error codes as the [reference implementation](https://github.com/graphql/express-graphql).
+The result of executing a GraphQL operation may contain partial data as well as encountered errors.
+Errors that happen during execution of the GraphQL operation become part of the result.
+
+If the server successfully executed parts of the operation and returns a well-formed response
+that contains a non-null key `data`, the status code SHOULD be `2xx`.
+
+If the status code is `200`, the response MUST be a well-formed GraphQL response.
+
+The server SHOULD respond with status code `400` (Bad Request) in case of a validation error that
+prevented execution of the request.
+
+The server SHOULD respond with the appropriate status code depending on the concrete error condition,
+for example `401` (Unauthorized) or `500` (Internal Server Error).
```

Comment:
These status codes should only apply if they are absolute (which most likely means the error occurred before the processing of the GraphQL request).  If an error occurs (either `401` or `500`) inside of a resolver or a directive's function, then a `200` should be sent, with an error key present.

Reply:
Agreed, i clarified that those status codes should only be returned for errors that completely prevent the successful execution of the request.